### ISSUE TITLE
[BUG] Installer issues

### ISSUE THREAD

**Issue by canihavesomecoffee** [CCExtractor/sample-platform#67]
**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [x] I have read and understood the [contributors guide](https://github.com/canihavesomecoffee/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [x] I have checked that the issue I'm posting isn't already reported.
- [x] I have checked that the issue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/canihavesomecoffee/sample-platform/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/canihavesomecoffee/sample-platform/issues)
- [x] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never visited/used the platform.
- [ ] I have used the platform just a couple of times.
- [ ] I have used the platform extensively, but have not contributed previously.
- [x] I am an active contributor to the platform.

---

- Installer can block on the installation of  nginx, python, pip, kvm, libvirt and virt-manager, when it's asking for the default keyboard mapping.
- easy_install of setuptools can fail. Better be using `pip install --upgrade setuptools`
- pip install can fail if libxml development libs are missing (libxml2-dev & libxslt1-dev).
- mysql_config_editor is not always present. We need to search for alternatives.

**Comment by thealphadollar** [comment]
in conjuction with #293

**Comment by thealphadollar** [comment]
Almost all of the above are fixed.

On successful installation of MySQL server, `mysql_config_editor` does come. Without mysql server, no point of having mysql_editor_config or any alternative.

**Comment by drepram** [comment]
Hi, sorry for coming out of nowhere, but as @thealphadollar said, shouldn't this issue be closed?

**Comment by thealphadollar** [comment]
@drepram Since there is always scope for improvement, not closing this issue isn't an issue at all. We are always looking for new ways to improve the installer.

**Comment by rdas6313** [comment]
Hello all, Since the issue is related to installer, i am posting it here instead of creating a new issue.

**Problem:-**

When the first time i was trying to install sample-platform in my local pc(Ubuntu 20.04) using installer script. then at the time of mysql install, Suddenly it stops for a few seconds then it creates an infinite loop of this message `Installing MySQL (root password will be empty!)`. Another thing to mention here, before starting the installer, i have installed virtualenv and also started it. To make sure it was not a coincidental incident, i went through the same process again (Tested it in a VM) and results were same as before.
Here's the[ log file.](https://pastebin.com/VuBcv19Q)

**How to replicate the problem:-**  

- First install virtualenv inside sample-platform or outside sample-platform but must be within www directory.
- Start virtualenv
- Goto sample-platform/install directory
- Start install.sh script
